src/test/java/org/osc/controller/nsc/InspectionHookApisTest.java

```diff
@@ -0,0 +1,267 @@
+                    .get();
+
+            response.close();
+
+            // Assert.
+            assertThat(response.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    @Test
+    public void testPostInspectionHook_expectStatusOk() {
+        // Assume.
+        Response response = null;
+        try {
+
+            InspectionHookEntity inspectionHook = getInspectionHookEntity();
+
+            Entity<InspectionHookEntity> entity = Entity.entity(inspectionHook, MediaType.APPLICATION_JSON);
+
+            // Act.
+            response = target("controller/1.2.3.0/inspectionHooks")
+                    .request()
+                    .header(this.authorizationHeader, this.authorizationCreds)
+                    .post(entity);
+
+            response.close();
+
+            // Assert.
+            assertThat(response.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    @Test
+    public void testUpdateInspectionHook_expectStatusOk() {
+        // Assume.
+        Response response = null;
+        try {
+
+            InspectionHookEntity inspectionHook = getInspectionHookEntity();
+
+            Entity<InspectionHookEntity> entity = Entity.entity(inspectionHook, MediaType.APPLICATION_JSON);
+            // Act.
+            response = target("controller/1.2.3.0/inspectionHooks/HookId")
+                    .request()
+                    .header(this.authorizationHeader, this.authorizationCreds)
+                    .put(entity);
+
+            response.close();
+
+            // Assert.
+            assertThat(response.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    @Test
+    public void testUpdateInspectionHook_withBadRequest_expectErrorCode() {
+        // Assume.
+        Response response = null;
+        try {
+
+            InspectionHookEntity inspectionHook = getInspectionHookEntity();
+
+            Entity<InspectionHookEntity> entity = Entity.entity(inspectionHook, MediaType.APPLICATION_JSON);
+
+            String badParam = "IdNotMatching";
+            // Act.
+            response = target("controller/1.2.3.0/inspectionHooks/" + badParam)
+                    .request()
+                    .header(this.authorizationHeader, this.authorizationCreds)
+                    .put(entity);
+
+            response.close();
+
+            // Assert.
+            assertThat(response.getStatus()).isEqualTo(500);
+
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    @Test
+    public void testDeleteInspectionHook_expectNoContent() {
```

Comment:
to be consistent with other project test names

testExecute_Action_ExpectedResult -> testExecute_WithDeleteInspectionHook_ExpectNoContent